traffic_ops/testing/api/v14/tenants_test.go

```diff
@@ -110,27 +114,195 @@ func DeleteTestTenants(t *testing.T) {
 	if err != nil {
 		t.Errorf("cannot GET Tenant by name: %v - %v\n", t1, err)
 	}
+	expectedChildDeleteErrMsg := `Tenant '` + strconv.Itoa(tenant1.ID) + `' has child tenants. Please update these child tenants and retry.`
+	if _, err := TOSession.DeleteTenant(strconv.Itoa(tenant1.ID)); err == nil {
+		t.Fatalf("%s has child tenants -- should not be able to delete", t1)
+	} else if !strings.Contains(err.Error(), expectedChildDeleteErrMsg) {
+		t.Errorf("expected error: %s;  got %s", expectedChildDeleteErrMsg, err.Error())
+	}
+
+	deletedTenants := map[string]struct{}{}
+	for {
```

Comment:
This makes `DeleteTestTenants` delete all tenants starting with children, instead of hard-coding which need deleting first.

Reply:
+1